keps/sig-node/4817-resource-claim-device-status/README.md

```diff
@@ -0,0 +1,727 @@
+      "User '" + request.userInfo.username + "' on node '" + variables.userNodeName + "' is not allowed to update the .status.devices of a ResourceClaim allocated on node '" + variables.objectNodeName + "'."
+    reason: Forbidden
+  variables:
+  - name: userNodeName
+    expression: >-
+      request.userInfo.extra[?'authentication.kubernetes.io/node-name'][0].orValue('')
+  - name: objectNodeName
+    expression: >-
+      object.status.allocation.nodeSelector.nodeSelectorTerms[0].matchFields[0].values[0].orValue('')
+---
+apiVersion: admissionregistration.k8s.io/v1
+kind: ValidatingAdmissionPolicyBinding
+metadata:
+  name: "resourceclaim-device-status-update-binding"
+spec:
+  policyName: "resourceclaim-device-status-update"
+  validationActions: [Deny]
+```
+
+### Test Plan
+
+[x] I/we understand the owners of the involved components may require updates to
+existing tests to make this code solid enough prior to committing the changes necessary
+to implement this enhancement.
+
+##### Prerequisite testing updates
+
+##### Unit tests
+
+- `ResourceClaim` validation:
+    - A device can only be reported once in the `ResourceClaim`.
+    - The reported device is allocated in the `ResourceClaim`.
+    - Properties set in `AllocatedDeviceStatus` are in the correct format.
+
+Coverage:
+- `k8s.io/kubernetes/pkg/apis/resource/validation`: `9/30/2024` - `77.1`
+
+##### Integration tests
+
+- Usage of the `Devices` field in the `ResourceClaimStatus`:
+    * With the feature gate enabled, the field exists in the `ResourceClaim`.
+    * With the feature gate disabled, the field does not exist in the
+      `ResourceClaim`.
+    * With the feature gate enabled, the `ValidatingAdmissionPolicy` exists and
+      restricts the write access of the `ResourceClaim.Status.Devices`.
+
+##### e2e tests
+
+TBD
+
+### Graduation Criteria
+
+#### Alpha
+
+- Feature implemented behind feature gates (`DRAResourceClaimDeviceStatus`).
+  Feature Gates are disabled by default.
+- Documentation provided.
+- Initial unit, integration and e2e tests completed and enabled.
+- Authorization implemented to allow only the user on the same node as the 
+  allocated `ResourceClaim` to write the status of the devices.
+
+#### Beta
+
+- Feature Gates are enabled by default.
+- No major outstanding bugs.
+- Feedback collected from the community (developers and users) with adjustments
+  provided, implemented and tested.
+
+#### GA
+
+- 2 examples of real-world usage.
+- Allowing time for feedback from developers and users.
+
+### Upgrade / Downgrade Strategy
+
+This feature only exposes a new field in the `ResourceClaim.Status`, the field
+will either be present or not.
+
+DRA implementation requires DRA interfaces change. DRA is in alpha and in
+active development. The feature will follow the DRA upgrade/downgrade strategy.
+
+### Version Skew Strategy
+
+This feature affects only the kube-apiserver, so there is no issue with version
+skew with other Kubernetes components.
```

Comment:
There could be skew with the DRA drivers themselves, but I guess that's their problem?